app/workers/question_worker.rb

```diff
@@ -0,0 +1,37 @@
+class QuestionWorker
+  include Sidekiq::Worker
+  def perform
+    require 'post_to_slack'
+
+    # First check there is no active question already
+    if Question.where(open: true).none?
+      # get a random unanswered question
+      today = Question.left_outer_joins(:responses).where(responses: {id: nil}).order("RANDOM()").first
```

Comment:
Do you need to do `left_outer_joins`? I feel like you might able to just do `joins` but I could be wrong.

Reply:
Could you look this up by `Question.where(date_asked: nil)` rather than loading the responses?

Reply:
Also, last thing about this, I would recommend calling your variable `todays_question` rather than `today` since it's a bit more explicit.

Reply:
I favored responses over `date_asked` in case the question is asked and never answered (such as on a holiday or something), so it can potentially be asked again. 🤔

Reply:
Not sure why, but `joins` doesn't work? It always returns nil.